File: sgi-frontend/src/types/analytics-comercial.ts
// =========================================================================
// RADAR COMERCIAL
// =========================================================================

export interface RadarKPIs {
  meta_ordenes_equipo: number;
  meta_individual: number;
  total_gestiones: number;
  clientes_nuevos: number;
}

export interface ProgresoComercial {
  comercial_id: number;
  nombre: string;
  iniciales: string | null;
  ordenes_logistico: number;
  ordenes_aduanas: number;
  ordenes_integral: number;
  total_ordenes: number;
  meta: number;
  porcentaje: number;
}

export interface PipelineComercial {
  comercial_id: number;
  nombre: string;
  prospectos: number;
  negociacion: number;
  cerrada: number;
  operacion: number;
  carga_entregada: number;
}

export interface AlertaClienteMuerto {
  cliente_id: number;
  cliente_nombre: string;
  comercial_id: number;
  comercial_nombre: string;
  dias_sin_contacto: number;
}

export interface RadarResponse {
  kpis: RadarKPIs;
  progreso_comerciales: ProgresoComercial[];
  pipeline_comerciales: PipelineComercial[];
  alertas_clientes_muertos: AlertaClienteMuerto[];
}

// =========================================================================
// EMBUDO Y DIAGNÓSTICO
// =========================================================================

export interface EtapaEmbudo {
  etapa: string;
  cantidad: number;
  porcentaje_retencion: number;
}

export interface ZeitPromedio { // Se mantiene por retrocompatibilidad
  etapa: string;
  dias_promedio: number;
}

export interface TiempoPromedio {
  etapa: string;
  dias_promedio: number;
}

export interface MotivoCaida {
  motivo: string;
  cantidad: number;
  porcentaje: number;
}

export interface EfectividadOrigen {
  origen: string;
  total_leads: number;
  cerrados: number;
  tasa_conversion: number;
}

export interface EmbudoResponse {
  embudo_conversion: EtapaEmbudo[];
  tiempos_promedio: TiempoPromedio[];
  motivos_caida: MotivoCaida[];
  efectividad_origen: EfectividadOrigen[];
}

// =========================================================================
// RENDIMIENTO COTIZACIONES KANBAN
// =========================================================================

export interface CotizacionesKpis {
  total_tarjetas: number;
  total_cotizaciones: number;
  total_ganadas: number;
  total_perdidas: number;
  tasa_conversion: number;
}

export interface ComercialCotizacionesRendimiento {
  comercial_id: number;
  nombre: string;
  iniciales: string | null;
  cotizados_creados: number;
  cierres_exitosos: number;
  negociaciones_caidas: number;
  tasa_efectividad: number;
  cotizaciones_pendientes: number;
  jefe_id?: number | null;
  jefe_nombre?: string | null;
}

export interface DistribucionCarga {
  tipo_carga_nombre: string;
  cantidad: number;
  porcentaje: number;
}

export interface DistribucionOperacion {
  tipo_operacion: string;
  cantidad: number;
  porcentaje: number;
}

export interface DistribucionSegmentacion {
  segmentacion_nombre: string;
  cantidad: number;
  porcentaje: number;
}

export interface TopMotivoCaida {
  motivo: string;
  cantidad: number;
  porcentaje: number;
}

export interface EmpresaCotizacionesRendimiento {
  cliente_id: number;
  nombre: string;
  cotizados_creados: number;
  cierres_exitosos: number;
  negociaciones_caidas: number;
  tasa_efectividad: number;
  cotizaciones_pendientes: number;
}

export interface CotizacionesAnalyticsResponse {
  kpis: CotizacionesKpis;
  rendimiento_comerciales: ComercialCotizacionesRendimiento[];
  rendimiento_empresas: EmpresaCotizacionesRendimiento[];
  distribucion_carga: DistribucionCarga[];
  distribucion_operacion: DistribucionOperacion[];
  distribucion_segmentacion: DistribucionSegmentacion[];
  motivos_caida: TopMotivoCaida[];
}

File: sgi-backend/app/schemas/comercial/analytics_comercial.py
from pydantic import BaseModel
from typing import List, Optional, Union

# =========================================================================
# RADAR COMERCIAL SCHEMAS
# =========================================================================

class RadarKPIs(BaseModel):
    meta_ordenes_equipo: int
    meta_individual: int
    total_gestiones: int
    clientes_nuevos: int


class ProgresoComercial(BaseModel):
    comercial_id: int
    nombre: str
    iniciales: Optional[str] = None
    ordenes_logistico: int
    ordenes_aduanas: int
    ordenes_integral: int
    total_ordenes: int
    meta: int
    porcentaje: float


class PipelineComercial(BaseModel):
    comercial_id: int
    nombre: str
    prospectos: int
    negociacion: int
    cerrada: int
    operacion: int
    carga_entregada: int


class AlertaClienteMuerto(BaseModel):
    cliente_id: int
    cliente_nombre: str
    comercial_id: int
    comercial_nombre: str
    dias_sin_contacto: int


class RadarResponse(BaseModel):
    kpis: RadarKPIs
    progreso_comerciales: List[ProgresoComercial]
    pipeline_comerciales: List[PipelineComercial]
    alertas_clientes_muertos: List[AlertaClienteMuerto]


# =========================================================================
# EMBUDO Y DIAGNÓSTICO SCHEMAS
# =========================================================================

class EtapaEmbudo(BaseModel):
    etapa: str
    cantidad: int
    porcentaje_retencion: float


class MotivoCaida(BaseModel):
    motivo: str
    cantidad: int
    porcentaje: float


class EfectividadOrigen(BaseModel):
    origen: str
    total_leads: int
    cerrados: int
    tasa_conversion: float


class TiempoPromedio(BaseModel):
    etapa: str
    dias_promedio: float


class EmbudoResponse(BaseModel):
    embudo_conversion: List[EtapaEmbudo]
    tiempos_promedio: List[TiempoPromedio]
    motivos_caida: List[MotivoCaida]
    efectividad_origen: List[EfectividadOrigen]


# =========================================================================
# RENDIMIENTO COTIZACIONES SCHEMAS
# =========================================================================

class CotizacionesKpis(BaseModel):
    total_cotizaciones: int          # Sum of all individual active Cotizacion records
    total_cargas_cotizadas: int      # Count of Seguimiento (tarjetas)
    total_clientes_gestionados: int  # Count DISTINCT cliente_id
    total_cierres: int               # Tarjetas in CIERRE + EN_OPERACION + CARGA_ENTREGADA
    total_en_operacion: int          # Tarjetas in EN_OPERACION + CARGA_ENTREGADA
    total_caidos: int                # Tarjetas in CAIDO
    tasa_conversion: float           # total_cierres / (total_cierres + total_caidos) * 100


class CotizacionDetalleSchema(BaseModel):
    cliente: str
    titulo: str
    tipo_carga: str
    servicio: str
    incoterm: str
    veces_cotizado: int
    estado: str


class ComercialCotizacionesRendimiento(BaseModel):
    comercial_id: int
    nombre: str
    iniciales: Optional[str] = None
    cotizaciones_totales: int       # SUM of active cotizaciones
    cargas_cotizadas: int           # COUNT of tarjetas
    clientes_gestionados: int       # COUNT DISTINCT of cliente_id
    cierres: int                    # Tarjetas in CIERRE + EN_OPERACION + CARGA_ENTREGADA
    en_operacion: int               # Tarjetas in EN_OPERACION + CARGA_ENTREGADA
    caidos: int                     # Tarjetas in CAIDO
    tasa_conversion: float          # cierres / (cierres + caidos) * 100
    jefe_id: Optional[int] = None
    jefe_nombre: Optional[str] = None
    detalle_cotizaciones: List[CotizacionDetalleSchema] = []


class DistribucionCarga(BaseModel):
    tipo_carga_nombre: str
    cantidad: int
    porcentaje: float


class DistribucionOperacion(BaseModel):
    tipo_operacion: str
    cantidad: int
    porcentaje: float


class DistribucionSegmentacion(BaseModel):
    segmentacion_nombre: str
    cantidad: int
    porcentaje: float


class TopMotivoCaida(BaseModel):
    motivo: str
    cantidad: int
    porcentaje: float


class EmpresaCotizacionesRendimiento(BaseModel):
    cliente_id: Union[int, str]
    nombre: str
    cotizaciones_totales: int
    cargas_cotizadas: int
    cierres: int
    en_operacion: int
    caidos: int
    tasa_conversion: float


class CotizacionesAnalyticsResponse(BaseModel):
    kpis: CotizacionesKpis
    rendimiento_comerciales: List[ComercialCotizacionesRendimiento]
    rendimiento_empresas: List[EmpresaCotizacionesRendimiento]
    distribucion_carga: List[DistribucionCarga]
    distribucion_operacion: List[DistribucionOperacion]
    distribucion_segmentacion: List[DistribucionSegmentacion]
    motivos_caida: List[TopMotivoCaida]
